libyul/backends/evm/EVMDialect.cpp

```diff
@@ -201,7 +201,7 @@ map<YulString, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _evmVe
 			"setimmutable",
 			2,
 			0,
-			SideEffects{false, false, false, false, true},
+			SideEffects{false, true, false, false, true, SideEffects::None, SideEffects::None, SideEffects::Write},
```

Comment:
This is a rather delicate opcode. It certainly writes to memory and just to be safe I would set it movableApartFromEffects: false